qutebrowser/browser/network/networkmanager.py

```diff
@@ -241,12 +242,23 @@ def clear_rejected_ssl_errors(self, url):
     @pyqtSlot('QNetworkReply', 'QAuthenticator')
     def on_authentication_required(self, reply, authenticator):
         """Called when a website needs authentication."""
-        answer = self._ask("Username ({}):".format(authenticator.realm()),
-                           mode=usertypes.PromptMode.user_pwd,
-                           owner=reply)
-        if answer is not None:
-            authenticator.setUser(answer.user)
-            authenticator.setPassword(answer.password)
+        user, password = None, None
+        if not hasattr(reply, "netrc_used"):
+            reply.netrc_used = True
+            net = netrc.netrc()
+            authenticators = net.authenticators(reply.url().host())
+            if authenticators is not None:
+                # pylint: disable=unpacking-non-sequence
+                (user, _account, password) = authenticators
+        if user is None:
+            answer = self._ask(
+                "Username ({}):".format(authenticator.realm()),
+                mode=usertypes.PromptMode.user_pwd,
+                owner=reply)
+            user, password = answer.user, answer.password
```

Comment:
I haven't tested this yet - but since I checked `if answer is not None:` in the old code, this seems wrong. What happens when you get the question and press esc?